endpoints/openrtb2/auction.go

```diff
@@ -156,15 +155,6 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 		return
 	}
 
-	resolvedFPD, fpdErrors := firstpartydata.ExtractFPDForBidders(req)
-	if len(fpdErrors) > 0 {
-		if errortypes.ContainsFatalError(fpdErrors) && writeError(fpdErrors, w, &labels) {
-			return
-		}
-		errL = append(errL, fpdErrors...)
-	}
-	warnings := errortypes.WarningOnly(errL)
-
```

Comment:
A couple of things here:
1) Do you recall what our line of thinking was on how to handle first party data errors? It looks like some errors are considered fatal and others are not. What types of errors fall into each of these categories?

2) It seems that by moving this logic down into `HoldAuction`, we might be returning a different error code when there is a fatal error. Here we were calling `writeError` which will return either a 400 or 503. When a fatal error occurs during `HoldAuction` FPD processing, it will be returned and handled in this function on line 226 which shows a 500 being returned:
```
if err != nil {
		labels.RequestStatus = metrics.RequestStatusErr
		w.WriteHeader(http.StatusInternalServerError)
		fmt.Fprintf(w, "Critical error while running the auction: %v", err)
		glog.Errorf("/openrtb2/auction Critical error: %v", err)
		ao.Status = http.StatusInternalServerError
		ao.Errors = append(ao.Errors, err)
		return
	}
```
I'm guessing a 500 doesn't make sense for most, if not all, of the FPD errors.

3) I'm trying to understand what the responsibility is of `Auction` vs `HoldAuction`. It seems like all request validation is done in `Auction` and any 400 errors would be caught there. In that case, it is probably not ideal to move `HoldAuction` down but I understand there are challenges with the other option we discussed so I'm ok with this strategy. I'm just pointing that out here since this just came to mind when I realized the potential http error status code discrepancy described in item 2.

Reply:
Error handling updated